pkg/crc/machine/cert-renewal.go

```diff
@@ -61,20 +59,9 @@ func (recoveryPod *recoveryPod) runPodCommand(cmd string) (string, error) {
 func startRecoveryPod(sshRunner *ssh.SSHRunner) (*recoveryPod, error) {
 	recoveryPod := recoveryPod{}
 	recoveryPod.sshRunner = sshRunner
-	recoveryPod.releaseImage = releaseImage
-
-	logging.Debugf("Using release image %s", recoveryPod.releaseImage)
-	kaoImage, err := sshRunner.Run(fmt.Sprintf("sudo oc adm release info --registry-config='/var/lib/kubelet/config.json' '%s' --image-for=cluster-kube-apiserver-operator", releaseImage))
-	if err != nil {
-		return nil, err
-	}
-	recoveryPod.kaoImage = strings.TrimSpace(kaoImage)
+	recoveryPod.kaoImage = kaoImage
 	logging.Debugf("kube-apiserver-operator image: %s", kaoImage)
 
-	_, err = sshRunner.Run(fmt.Sprintf("sudo podman pull --authfile='/var/lib/kubelet/config.json' %s", kaoImage))
```

Comment:
With these references to `config.json` gone, some of the machine.go changes introduced by `3f84454e machine: Trigger certificate renewal when they are expired` can probably be removed:
```
diff --git a/pkg/crc/machine/machine.go b/pkg/crc/machine/machine.go
index b550443c..582211aa 100644
--- a/pkg/crc/machine/machine.go
+++ b/pkg/crc/machine/machine.go
@@ -341,12 +341,6 @@ func Start(startConfig StartConfig) (StartResult, error) {
 			result.Error = err.Error()
 			return *result, errors.Newf("Error copying kubeconfig file  %v", err)
 		}
-
-		err = pullsecret.AddPullSecretToInstanceDisk(sshRunner, pullSecret)
-		if err != nil {
-			result.Error = err.Error()
-			return *result, errors.Newf("Failed to update user pull secret or cluster ID: %v", err)
-		}
 	}
 
 	if needsCertsRenewal {
diff --git a/pkg/crc/pullsecret/pullsecret.go b/pkg/crc/pullsecret/pullsecret.go
index 81ee535e..1c40a81c 100644
--- a/pkg/crc/pullsecret/pullsecret.go
+++ b/pkg/crc/pullsecret/pullsecret.go
@@ -62,7 +62,7 @@ func AddPullSecretAndClusterID(sshRunner *ssh.SSHRunner, pullSec string, kubecon
 	}
 
 	// Add user pull secret to the instance
-	if err := AddPullSecretToInstanceDisk(sshRunner, pullSec); err != nil {
+	if err := addPullSecretToInstanceDisk(sshRunner, pullSec); err != nil {
 		return err
 	}
 	return nil
@@ -78,7 +78,7 @@ func addpullSecretSpecToInstance(sshRunner *ssh.SSHRunner, pullSec string) error
 	return nil
 }
 
-func AddPullSecretToInstanceDisk(sshRunner *ssh.SSHRunner, pullSec string) error {
+func addPullSecretToInstanceDisk(sshRunner *ssh.SSHRunner, pullSec string) error {
 	output, err := sshRunner.Run(fmt.Sprintf("cat <<EOF | sudo tee /var/lib/kubelet/config.json\n%s\nEOF", pullSec))
 	if err != nil {
 		return err
```

Reply:
@cfergeau yes we can revert those, will put another commit to revert it back.